enhancements/kube-apiserver/stability-api-removal-notifications.md

```diff
@@ -112,9 +134,66 @@ Value of `removed_release`:
 | 4.9               | 1.23            |                       |
 | 4.10 / 4.10 EUS   | 1.24            | TBD                   |
 
+#### DeprecatedAPIRequest Resource
+
+Patch the apiserver such that requests to a deprecated API are logged to a `DeprecatedAPIRequest` resource corresponding
+to the deprecated API. This resource can help customers determine which workloads are using deprecated APIs, including
+those which are a being removed in the next release.
+
+```go
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// DeprecatedAPIRequest tracts requests made to a deprecated API. The instance name should
+// be of the form `resource.group.version`, matching the deprecated resource.
+type DeprecatedAPIRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Status DeprecatedAPIRequestStatus `json:"status,omitempty"`
+}
+
+type DeprecatedAPIRequestStatus struct {
+
+	// RemovedRelease is when the API will be removed.
+	RemovedRelease string `json:"removedRelease"`
+
+	// Sources from which users have accessed the deprecated API.
+	Sources []DeprecatedAPIRequestSource `json:"sources,omitempty"`
+}
+
+// DeprecatedAPIRequestSource represents a source, identified by IP, from which a deprecated
+// API request was made.
+type DeprecatedAPIRequestSource struct {
+
+	// IP address of the source.
+	IP string `json:"ip"`
+
+	// Users list the last ten unique users to request the deprecated API from this source
```

Comment:
this was my first thought, but I'm not sure how we properly handle server restarts in this hour correctly